### DISCUSSION TITLE
How do I setup authentication with Google SSO?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#523] [Q&A] [answered]
Hi,

I saw that we can configure authentication with Google: 
https://docs.atoti.io/0.6.4/lib/atoti.config.authentication.oidc.html#atoti.config.authentication.oidc.OidcConfig

Do you have any examples of how we can configure it? Thanks

**Comment by bloussou** [comment, marked as answer]
Hello,
It is the same idea as the ["how to secure a session"](https://docs.atoti.io/latest/how_tos/security/security.html#Configuring-the-authentication-mechanism).
For google it looks like : 
```python
"authentication": {
    "oidc": {
        "provider_id": "google",
        "issuer_url": "https://accounts.google.com",
        "client_id": CLIENT_ID,
        "client_secret": CLIENT_SECRET,
        "name_claim": "email",
        "scopes": ["https://www.googleapis.com/auth/userinfo.email"],
        "role_mapping": {
            "ROLE_USER": ["ROLE_USER", "ROLE_SHARE"],
        },
    }
},
```
Secret and ID can be setup following [`oidc google documentation`](https://developers.google.com/identity/protocols/oauth2/openid-connect#getcredentials)

**Reply by tibdex** [reply]
Because Google doesn't seem to support sending roles in its OIDC tokens, you might need to wait for the next release and its introduction of OIDC default roles to be able to grant the required `ROLE_USER` to all users connecting through Google.
